src/metabase/query_processor/middleware/annotate.clj

```diff
@@ -126,6 +126,10 @@
 
 ;;; --------------------------------------------------- Field Info ---------------------------------------------------
 
+(defn- name-for-joined-field
```

Comment:
Note that this is plain wrong to do it in all cases, but just "appears" to give the expected result in some cases. A proper solution should be implemented